dGame/dComponents/RocketLaunchLupComponent.cpp

```diff
@@ -0,0 +1,55 @@
+#include "RocketLaunchLupComponent.h"
+#include "CDClientDatabase.h"
+#include "RocketLaunchpadControlComponent.h"
+#include "CharacterComponent.h"
+#include "Item.h"
+
+RocketLaunchLupComponent::RocketLaunchLupComponent(Entity* parent) : Component(parent) {
+	m_Parent = parent;
+
+	// get the lup worlds from the cdclient
+	std::string query = "SELECT * FROM LUPZoneIDs;";
+	auto results = CDClientDatabase::ExecuteQuery(query);
+	while (!results.eof()) {
+		// fallback to 1600 incase there is an issue
+		m_LUPWorlds.push_back(results.getIntField(0, 1600));
+		results.nextRow();
+	}
+	results.finalize();
+}
+
+RocketLaunchLupComponent::~RocketLaunchLupComponent() {
+	if (!m_rocket) return;
+	// when we exit the ui we need to unequip the rocket
+	m_rocket->UnEquip();
```

Comment:
Believe you misunderstand when this will be called - this is in the component destructor, this will only ever be called when the rocket launcher itself is destroyed.